todo-list-ui/src/app/todo-modal/todo-modal.component.ts

```diff
@@ -0,0 +1,42 @@
+import { FormControl, ReactiveFormsModule } from '@angular/forms';
+import {
+  Component,
+  inject,
+  signal,
+  TemplateRef,
+  WritableSignal,
+} from '@angular/core';
+
+import {
+  NgbDatepickerModule,
+  NgbModal,
+  NgbModalRef,
+} from '@ng-bootstrap/ng-bootstrap';
+import { todos } from '../../mocks/todos.mocks';
+
+@Component({
+  selector: 'app-todo-modal',
+  standalone: true,
+  imports: [ReactiveFormsModule, NgbDatepickerModule],
+  templateUrl: './todo-modal.component.html',
+})
+export class TodoModalComponent {
+  private modalService = inject(NgbModal);
+  private modalInstance?: NgbModalRef;
+  closeResult: WritableSignal<string> = signal('');
+  inputTodo = new FormControl('', { nonNullable: true });
+
+  open(content: TemplateRef<any>) {
+    this.modalInstance = this.modalService.open(content);
+  }
+
+  add() {
```

Comment:
As a good practice. Explicitly add the return type to every method or function you define